server/src/main/java/org/opensearch/action/bulk/BulkRequestParser.java

```diff
@@ -176,7 +203,7 @@ public void parse(
                             + "]"
                     );
                 }
-                String action = parser.currentName();
+                Action action = Action.of(parser.currentName(), line);
```

Comment:
You could maybe refactor this as `Action.consume(parser, line)` so that you can handle the failure case above in the same place. I think it currently prints "...expected FIELD_NAME but found [x]" and that could be improved as well to say the "expected one of ..." thing.

Reply:
`consume` sounds more like it doesn't return anything, let me think of a better name.
the error message does say `... expected one of [create, delete, index, update] but found ...`

Reply:
Yes, your new error message says that. I was suggesting to make it say that for the case on line 195 where a non-field name is encountered as well.

`Action.parse` might work.

Reply:
Got it, thanks!